caas/kubernetes/provider/modeloperator.go

```diff
@@ -400,6 +502,127 @@ func modelOperatorService(
 	}
 }
 
+func ensureModelOperatorRBAC(
```

Comment:
I think we're missing rollback / cleanup of resources if something fails part way through?

Reply:
Yes because we don't expect this to fail. Should we add it as we don't have it around a lot of the other code paths.

Reply:
The other k8s ensure methods have cleanup so would be good